elasticdl/python/master/elasticdl_job_service.py

```diff
@@ -78,52 +95,25 @@ def __init__(self, args, task_manager, rendezvous_server=None):
             )
         )
 
-        # Initialize instance manager
-        self.instance_manager = self._create_instance_manager(args)
+        # Initialize pod manager
+        self.pod_manager = pod_manager
 
         self.task_manager.set_task_timeout_callback(
-            self.instance_manager._remove_worker
+            self.pod_manager._remove_worker
         )
 
     def start(self):
         """
         Start the components one by one. Make sure that it is ready to run.
         """
         # Start the worker manager if requested
-        if self.instance_manager:
-            self.instance_manager.update_status(InstanceManagerStatus.PENDING)
+        if self.pod_manager:
+            self.pod_manager.update_status(InstanceManagerStatus.PENDING)
             if self.num_ps_pods > 0:
                 self.logger.info("num ps pods : {}".format(self.num_ps_pods))
-                self.instance_manager.start_parameter_servers()
-            self.instance_manager.start_workers()
-            self.instance_manager.update_status(InstanceManagerStatus.RUNNING)
-
-    @staticmethod
-    def _get_job_type(args):
-        if all(
-            (args.training_data, args.validation_data, args.evaluation_steps,)
-        ):
-            job_type = JobType.TRAINING_WITH_EVALUATION
-        elif all(
-            (
-                args.validation_data,
-                not args.training_data,
-                not args.prediction_data,
-            )
-        ):
-            job_type = JobType.EVALUATION_ONLY
-        elif all(
-            (
-                args.prediction_data,
-                not args.validation_data,
-                not args.training_data,
-            )
-        ):
-            job_type = JobType.PREDICTION_ONLY
-        else:
-            job_type = JobType.TRAINING_ONLY
-
-        return job_type
+                self.pod_manager.start_parameter_servers()
+            self.pod_manager.start_workers()
+            self.pod_manager.update_status(InstanceManagerStatus.RUNNING)
```

Comment:
Can we move the pod_manager instance out of elasticdl job service later?

Reply:
Yes, we can.
Will move the instance out in the next PR.